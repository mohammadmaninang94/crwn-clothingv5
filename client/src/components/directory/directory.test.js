import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import Directory from './directory.component';
import sections from '../../assests/data/sections.data';

describe('Directory component', () => {
    describe('Renders with', () => {
        let wrapper = null;

        beforeEach(() => {
            wrapper = render(
                <MemoryRouter>
                    <Directory sections={sections} />
                </MemoryRouter>
            );
        });

        test(`exact titles from props`, () => {
            for (var i = 0; sections.length > i; i++) {
                expect(wrapper.queryByTitle(sections[0].title)).toBeInTheDocument();
            }
        });

        test(`${sections.length} menu item component`, () => {
            expect(wrapper.queryAllByRole('heading')).toHaveLength(sections.length);
        });
    });

    test('renders correctly', () => {
        const tree = renderer
            .create(<MemoryRouter>
                <Directory sections={sections} />
            </MemoryRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});