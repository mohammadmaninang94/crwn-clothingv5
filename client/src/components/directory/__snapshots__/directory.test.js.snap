// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory component renders correctly 1`] = `
<div
  className="sc-jSgupP juiZeu"
>
  <div
    className="sc-dlfnbm jKujKF"
    onClick={[Function]}
    title="hats"
  >
    <div
      className="sc-bdfBwQ jzvLqE"
    />
    <div
      className="sc-gsTCUz beQqUW"
    >
      <h2
        className="sc-hKgILt jwqUMt"
      >
        hats
      </h2>
      <span
        className="sc-eCssSg eQuoXW"
      >
        Shop Now
      </span>
    </div>
  </div>
  <div
    className="sc-dlfnbm gZJZwt"
    onClick={[Function]}
    title="jackets"
  >
    <div
      className="sc-bdfBwQ hPGIbn"
    />
    <div
      className="sc-gsTCUz beQqUW"
    >
      <h2
        className="sc-hKgILt jwqUMt"
      >
        jackets
      </h2>
      <span
        className="sc-eCssSg eQuoXW"
      >
        Shop Now
      </span>
    </div>
  </div>
  <div
    className="sc-dlfnbm kMCrCK"
    onClick={[Function]}
    title="sneakers"
  >
    <div
      className="sc-bdfBwQ exgQJA"
    />
    <div
      className="sc-gsTCUz beQqUW"
    >
      <h2
        className="sc-hKgILt jwqUMt"
      >
        sneakers
      </h2>
      <span
        className="sc-eCssSg eQuoXW"
      >
        Shop Now
      </span>
    </div>
  </div>
  <div
    className="sc-dlfnbm cUYBAg"
    onClick={[Function]}
    title="womens"
  >
    <div
      className="sc-bdfBwQ fuJqLj"
    />
    <div
      className="sc-gsTCUz beQqUW"
    >
      <h2
        className="sc-hKgILt jwqUMt"
      >
        womens
      </h2>
      <span
        className="sc-eCssSg eQuoXW"
      >
        Shop Now
      </span>
    </div>
  </div>
  <div
    className="sc-dlfnbm dmWhCB"
    onClick={[Function]}
    title="mens"
  >
    <div
      className="sc-bdfBwQ hgLyjY"
    />
    <div
      className="sc-gsTCUz beQqUW"
    >
      <h2
        className="sc-hKgILt jwqUMt"
      >
        mens
      </h2>
      <span
        className="sc-eCssSg eQuoXW"
      >
        Shop Now
      </span>
    </div>
  </div>
</div>
`;
